generators/test/templates/template_test.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestTerraformSshExample(t *testing.T) {
+	t.Parallel()
+
+	exampleFolder := "./fixture"
+
+	// At the end of the test, run `terraform destroy` to clean up any resources that were created
+	defer test_structure.RunTestStage(t, "teardown", func() {
+		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+		terraform.Destroy(t, terraformOptions)
+	})
+
+	// Deploy the example
+	test_structure.RunTestStage(t, "setup", func() {
+		terraformOptions := configureTerraformOptions(t, exampleFolder)
+
+		// Save the options so later test stages can use them
+		test_structure.SaveTerraformOptions(t, exampleFolder, terraformOptions)
+
+		// This will run `terraform init` and `terraform apply` and fail the test if there are any errors
+		terraform.InitAndApply(t, terraformOptions)
+	})
+
+	// Check whether the output has a length of 22
+	test_structure.RunTestStage(t, "validate", func() {
+		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+
+		stringList := terraform.Output(t, terraformOptions, "permutation_string_list_test")
+		fmt.Println(stringList)
+		if len(stringList) != 22 {
```

Comment:
Use constant to organize your numbers.

Reply:
This number varies based on the testing example. Here using constant is OK but need to be modified or even deleted if we use other test cases.

Reply:
Shall we put these contansts at the beginning of the code?

---
In reply to: [208440703](https://github.com/Azure/generator-az-terra-module/pull/2#discussion_r208440703) [](ancestors = 208440703)

Reply:
Because this constant is related to test case, it may be a better idea to put it near core test code (checking length of output here).